### PR TITLE
feat: add check of minimal recommended zksolc version

### PR DESCRIPTION
**Pull request by Velnbur**
# What :computer: 
* Add fetching of recommended minimal `zsksolc` version as mentioned in #96 
* Add check for version passed through `--use-zksolc` parameter that it's greater than minimal
* Add an ability to skip this check using `ZKSOLC_SKIP_MINIMAL_VERSION_CHECK`
# Why :hand:
* Closes: #96 

# Evidence :camera:

image

### CODE REVIEW

**Comment by Velnbur** [comment]
@dutterbutter my bad, fixed the `clippy` warnings.

Also I may do the #208 and #42 here too, as they are related

**Comment by dutterbutter** [comment]
@Velnbur thanks for the contribution! If you intend to pick up those issues please do so in a separate PR.

File: crates/common/src/zksolc_manager.rs [hunk 1]
```diff
@@ -674,3 +725,77 @@ impl ZkSolcManager {
...
+    pub min_version: ZkSolcVersion,
+}
+
+impl TryFrom<SolcMetadataResponseBody> for SolcMetadata {
+    type Error = anyhow::Error;
+
+    fn try_from(value: SolcMetadataResponseBody) -> Result<Self, Self::Error> {
+        Ok(SolcMetadata {
+            // HACK: the `version.json` file has version without `v` prefix, but
+            // `parse_version` requires them, we so we add them this `format!`.
```
Comment (dutterbutter) [inline crates/common/src/zksolc_manager.rs:796]: @Velnbur can you please update comment to indicate a `TODO` please. Otherwise this LGTM!

Comment (Velnbur) [reply crates/common/src/zksolc_manager.rs]: done

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**